src/LineageTree/lineageTree.py

```diff
@@ -2223,6 +2223,7 @@ def get_cells_at_t_from_root(self, r: [int, list], t: int = None) -> list:
                     to_do.append(_next)
                 elif self.time[_next] == t:
                     final_nodes.append(_next)
+        if not final_nodes: return list(r)
```

Comment:
I see the problem.
Though I am not sure why you are making a copy of `r`?

I made an imprecision line 2215: I check if `r` is a `list` but it could be a `set` or a `np.ndarray` and it would work also and we probably want to accept `set`s and `ndarray`s. One way around it would be to test if `r` can be iterated over:
```python
if not hasattr(r, "__iter__"):
  r = [r]
```

Also, for information, another way to handle the problem mentioned above (the empty list) would be to rewrite slightly the function (see below)
In fact my proposed solution would probably be even more versatile since it allows to only return cells at time `t` if initially not all the cells in `r` are at time `t`. Moreover, it does not return `r` if the lineage tree does not contain any cell at time `t`.
```python
def get_cells_at_t_from_root(self, r: [int, list], t: int = None) -> list:
  """
  Returns the list of cells at time `t` that are spawn by the node(s) `r`.
      Args:
          r (int | list): id or list of ids of the spawning node
          t (int): target time, if None goes as far as possible
                  (default None)
      Returns:
          (list) list of nodes at time `t` spawned by `r`
  """
  if not hasattr(r, "__iter__"):
    r = [r]
  to_do = list(r)
  final_nodes = []
  while len(to_do) > 0:
    curr = to_do.pop()
    if self.time[curr] == t:
      final_nodes.append(curr)
    elif self.time[curr] < t:
      to_do.extend(self[curr])
  return final_nodes
```